lib/topics_filter.rb

```diff
@@ -92,110 +77,207 @@ def filter_status(status:, category_id: nil)
 
   private
 
-  def filter_created_by_user(usernames:)
-    register_scope(
-      key: :created_by_user,
-      params: {
-        usernames: usernames.map(&:downcase),
-      },
-    ) do |usernames_lower|
-      @scope.joins(:user).where("users.username_lower IN (?)", usernames_lower)
+  def filter_categories(values:)
+    exclude_subcategories_category_slugs = []
+    include_subcategories_category_slugs = []
+
+    values.each do |key_prefix, value|
+      break if key_prefix && key_prefix != "="
+
+      value
+        .scan(
+          /^(?<category_slugs>([a-zA-Z0-9\-:]+)(?<delimiter>[,])?([a-zA-Z0-9\-:]+)?(\k<delimiter>[a-zA-Z0-9\-:]+)*)$/,
```

Comment:
Unrelated to this PR, but we have the `slug_generation_method` setting which can be set to `encoded` to percent-encode non-ASCII characters in category slugs. If I'm reading this regex correctly, I don't think it can match an encoded slug, but we should test to confirm at some point.

Reply:
Yea it definitely won't be able to because I'm restricting it to a certain set of chars. Thank you for letting me know, I'll address this in another PR.